katti/webkattis.py

```diff
@@ -261,6 +417,47 @@ def login(kattis_config: configloader.KattisConfig, verbose=False) -> requests.R
         r.raise_for_status()
     return r
 
+def login_with_password(kattis_config: configloader.KattisConfig, verbose=False) -> requests.Response:
+    """
+    A helper function to log a user in to kattis
+
+    Parameters
+    ----------
+    kattis_config: configloader.KattisConfig
+        A KattisConfig object containing the user's login credentials
+
+    Returns
+    -------
+    requests.Response
+        A response object containing the response from the login request
+    """
+    global kattis_session
+
+    # clear session
+    kattis_session = requests.Session()
+
+    r = kattis_session.get(f'{kattis_config.url}{_LOGIN_ENDING}/email', headers=_HEADERS)
+    regex_result = re.findall(r'value="(\d+)"', r.text)
+    csrf_token = regex_result[0]
+    data = {
+            'csrf_token': csrf_token,
+            'user': kattis_config.username,
+            'password':kattis_config.password #TODO: change to something more secure
```

Comment:
It's not in the URL, which I think is the worst thing. This might be ok.

Reply:
yeah I agree